Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/ChangeEvent.swift

```diff
@@ -179,14 +179,15 @@ public struct ChangeEvent<DocumentT: Codable>: Codable, Hashable {
     static func changeEventForLocalReplace(namespace: MongoNamespace,
                                            documentId: BSONValue,
                                            document: Document,
+                                           updateDescription: UpdateDescription? = nil,
```

Comment:
Replace events don't contain update descriptions per the MongoDB change event spec
https://docs.mongodb.com/manual/reference/change-events/#replace-event

When you say the update description contains vital information, do you mean vital to the end user or vital for the synchronization to happen properly? If it's the latter, then I think our coalescence algorithm is incorrect. What kind of information is "vital"? 

Seems like a user should be able to find everything they need to know from the fullDocument. If they need to see the update description, then maybe we shouldn't be coalescing the events in the first place. thoughts @edaniels @dkaminsky ?

Reply:
For my use case, I'm updating the `index` of a `TodoItem`. This is to make sure that all TodoLists are ordered the same. Kind of a handy feature, and the todo apps I use definitely sync ordering.

There are specific actions I want to take in the event that an `index` changes– namely, to change the ordering of the todoList. For this purpose, I have an `index_swap` table– a single document keyed on `owner_id` that tells me which two elements to swap on the list. However, I'm still going to receive the update for the index in the `items` collection. While I want to update the internal list with the new information, I do _not_ want to update the tableView, as it's being updated from the `index_swap`. So I check for whether or not the `index` field has been updated and if it has, I do not update the tableView. Now, I could discern this via running my own diff, but that seems unnecessary.

This seems like it would be a fairly common use case– not for todo lists per se, but most updates are going to be coalesced to replaces. Which means most update descriptions are going to be written off.

Reply:
Hmm, so I personally think it will probably be a best practice for users to do their own diff since change events my come from various sources outside of their own app, that won't be guaranteed to be an update (or vice versa).

though I get where you're coming from, and it is somewhat jarring for a simple app where the user knows all the events they're doing. Perhaps it would make sense for coalescence to be a configurable setting? 

I get that having the update description in the replace will be useful, but then we are diverging from something standard that MongoDB does, and it will set up an expectation from users that all replace events will have an updatedescription, when that is not the case. Replace events coming from the Stitch server will have no update description.

@dkaminsky @edaniels what are your thoughts on making change event coalescence a toggle-able setting?